lib/ui/home/tabs/locker_tab.dart

```diff
@@ -0,0 +1,455 @@
+import 'package:cobble/domain/apps/app_compatibility.dart';
+import 'package:cobble/domain/apps/app_manager.dart';
+import 'package:cobble/domain/connection/connection_state_provider.dart';
+import 'package:cobble/domain/db/models/app.dart';
+import 'package:cobble/domain/entities/hardware_platform.dart';
+import 'package:cobble/localization/localization.dart';
+import 'package:cobble/ui/common/components/cobble_button.dart';
+import 'package:cobble/ui/common/components/cobble_divider.dart';
+import 'package:cobble/ui/common/components/cobble_fab.dart';
+import 'package:cobble/ui/common/components/cobble_sheet.dart';
+import 'package:cobble/ui/common/components/cobble_tile.dart';
+import 'package:cobble/ui/common/icons/fonts/rebble_icons.dart';
+import 'package:cobble/ui/router/cobble_scaffold.dart';
+import 'package:cobble/ui/router/cobble_screen.dart';
+import 'package:cobble/ui/theme/with_cobble_theme.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_svg_provider/flutter_svg_provider.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:share/share.dart';
+
+class LockerTab extends HookWidget implements CobbleScreen {
+  @override
+  Widget build(BuildContext context) {
+    final connectionState = useProvider(connectionStateProvider.state);
+
+    final currentWatch = connectionState.currentConnectedWatch;
+
+    final appManager = useProvider(appManagerProvider);
+    List allPackages = useProvider(appManagerProvider.state);
+    List incompatibleApps =
+        allPackages.where((element) => !element.isWatchface).toList();
+    List incompatibleFaces =
+        allPackages.where((element) => element.isWatchface).toList();
+    List compatibleApps = [];
+    List compatibleFaces = [];
+    WatchType watchType;
+    bool circleConnected = false;
+    PebbleWatchLine lineConnected = PebbleWatchLine.unknown;
+
+    if (currentWatch != null) {
+      watchType = currentWatch.runningFirmware.hardwarePlatform.getWatchType();
+      circleConnected = watchType == WatchType.chalk;
+      lineConnected = currentWatch.line;
+      compatibleApps = incompatibleApps
+          .where((element) =>
+              AppCompatibility(element).isCompatibleWith(watchType))
+          .toList();
+      compatibleFaces = incompatibleFaces
+          .where((element) =>
+              AppCompatibility(element).isCompatibleWith(watchType))
+          .toList();
+      incompatibleApps = incompatibleApps
+          .where((element) =>
+              !AppCompatibility(element).isCompatibleWith(watchType))
+          .toList();
+      incompatibleFaces = incompatibleFaces
+          .where((element) =>
+              !AppCompatibility(element).isCompatibleWith(watchType))
+          .toList();
+    }
+
+    Widget facesPreview(App face, bool compatible, bool extended) {
```

Comment:
Would it be possible to extract these methods into separate Widgets/files? This file is almost 500 lines long.